sup3r/bias/bias_calc.py

```diff
@@ -414,6 +390,109 @@ def get_base_data(base_fps, base_dset, base_gid, base_handler,
+        Parameters
+        ----------
+        res : rex.Resource
+            rex Resource handler that is an open file handler of the base
+            file(s)
+        base_dset : str
+            A single dataset from the base_fps to retrieve.
+        base_gid : int | np.ndarray
+            One or more spatial gids to retrieve from base_fps. The data will
+            be spatially averaged across all of these sites.
+
+        Returns
+        -------
+        base_data : np.ndarray
+            1D array of base data spatially averaged across the base_gid input
+        base_cs_ghi : np.ndarray | None
+            If base_dset == "clearsky_ratio", the base_data array is GHI and
+            this base_cs_ghi is clearsky GHI. Otherwise this is None
+        """
+
+        base_cs_ghi = None
+
+        if base_dset.startswith(('U_', 'V_')):
+            dset_ws = base_dset.replace('U_', 'windspeed_')
+            dset_ws = dset_ws.replace('V_', 'windspeed_')
+            dset_wd = dset_ws.replace('speed', 'direction')
+            base_ws = res[dset_ws, :, base_gid]
+            base_wd = res[dset_wd, :, base_gid]
+
+            if base_dset.startswith('U_'):
+                base_data = -base_ws * np.sin(np.radians(base_wd))
+            else:
+                base_data = -base_ws * np.cos(np.radians(base_wd))
+
+        elif base_dset == 'clearsky_ratio':
+            base_data = res['ghi', :, base_gid]
+            base_cs_ghi = res['clearsky_ghi', :, base_gid]
+
+        else:
+            base_data = res[base_dset, :, base_gid]
+
+        if len(base_data.shape) == 2:
+            base_data = base_data.mean(axis=1)
+            if base_cs_ghi is not None:
+                base_cs_ghi = base_cs_ghi.mean(axis=1)
+
+        return base_data, base_cs_ghi
+
+    @staticmethod
+    def _reduce_base_data(base_ti, base_data, base_cs_ghi, base_dset,
+                          daily_reduction):
+        """Reduce the base timeseries data using some sort of daily reduction
+        function.
+
+        Parameters
+        ----------
+        base_ti : pd.DatetimeIndex
+            Time index associated with base_data
+        base_data : np.ndarray
+            1D array of base data spatially averaged across the base_gid input
+        base_cs_ghi : np.ndarray | None
+            If base_dset == "clearsky_ratio", the base_data array is GHI and
+            this base_cs_ghi is clearsky GHI. Otherwise this is None
+        base_dset : str
+            A single dataset from the base_fps to retrieve.
+        daily_reduction : str
+            Option to do a reduction of the hourly+ source base data to daily
+            data. Can be None (no reduction, keep source time frequency), "avg"
+            (daily average), "max" (daily max), or "min" (daily min)
+
+        Returns
+        -------
+        base_data : np.ndarray
+            1D array of base data spatially averaged across the base_gid input
+            and possibly daily-averaged as well.
```

Comment:
I suppose this could be daily-averaged or max/min. Everything looks good so I had to find one thing to nitpick.

Reply:
I think you're saying it should say min/max'd in addition to daily-averaged? Makes sense to me! added along with a real functional fix.